src/zjs_timers.c

```diff
@@ -108,9 +107,19 @@ static bool delete_timer(int32_t id)
     return false;
 }
 
-static void zjs_timer_free_cb(uintptr_t handle)
+void zjs_timers_free_timers()
 {
-    delete_timer(((zjs_timer_t *)handle)->callback_id);
+    while (zjs_timers) {
+        zjs_timer_t *tm = zjs_timers;
+        zjs_port_timer_stop(&tm->timer);
+        zjs_timers = tm->next;
```

Comment:
I'd move this to the bottom of the while() to make it more clear that this is the loop incrementer. Sorry for a second pass of comments, but they're quick ones. :)